### DISCUSSION TITLE
Release pkgs / rosdep for abb_robot_driver and associated pkgs

### DISCUSSION THREAD

**Discussion by BenArtes** [ros-industrial/abb_robot_driver#51] [Q&A] [answered]
Is there a plan or timeline for releasing the abb_robot_driver and associated (abb_libegm, abb_librws, abb_egm_rws_managers) debian packages and adding them to rosdep?

**Comment by gavanderhoorn** [comment, marked as answer]
There are currently no plans to release binaries for the packages you mention.

Note that [abb_robot_driver_interfaces](https://github.com/ros-industrial/abb_robot_driver_interfaces) (ie: the `.msg`s and `.srv`s) has been released.

> adding them to rosdep

a bit of a nit, but: packages are not added to `rosdep` when they are released. `rosdep` is intelligent enough to figure out something is a ROS package, and takes that information from the `ros/rosdistro`. "Adding to `rosdep`" is typically taken to mean to add a "key" to the rosdep DB, which would be for system dependencies only.

Could you perhaps clarify what you're trying to accomplish / avoid? Perhaps we can offer an alternative.

**Comment by BenArtes** [comment]
Partially just so that using them follows the workflow of other ROS dependencies. List them in package.xml and use rosdep to install all the deps of our projects.

The main reason though is an awkward paint point caused by using CLion. Essentially only building with `catkin_make` is supported because it creates a top-level CMakeLists which allows CLion to load an entire workspace. They have an [issue](https://youtrack.jetbrains.com/issue/CPP-7438) tracking official support (though not sure if / when it will be addressed). I also just noticed a recent [ROS2 setup guide](https://www.jetbrains.com/help/clion/ros2-tutorial.html#build-pkg-clion) that relies on the complication database.

The non-catkin CMake dependencies (abb_libegm, abb_librws, etc) force us to have to use an isolated build / catkin_tools which aren't really supported by CLion.

Obviously relying on an old non-isolated build tool for IDE support isn't ideal. We use catkin_tools in CI to ensure that our builds don't rely on a lack of isolation.

That said, I found a solution that works for us by building and installing abb_robot_driver + deps  to a folder outside of src and overlaying it onto our workspace.
